src/ast/desugar.rs

```diff
@@ -257,16 +277,48 @@ fn flatten_actions(actions: &Vec<Action>, desugar: &mut Desugar) -> Vec<NormActi
                 let added_variants = add_expr(variants.clone(), &mut res);
                 res.push(NormAction::Extract(added, added_variants));
             }
+            // TODO: Reduce duplication in these three cases
```

Comment:
I think you can do this TODO by matching on `Action::Delete(symbol, exprs) | Action::Unextractable(symbol, exprs)...`

Reply:
Yeah but then how do I differentiate which branch it went down to match to decide which `ChangeRow` enum to use?